### ISSUE TITLE
Account Filter Dictionary Serializes Enum Name instead of EnumMember value

### ISSUE THREAD

**Issue by grcodemonkey** [viceroypenguin/Going.Plaid#37]
When serializing the AccountFilters property on CreateLinkTokenRequest, the AccountType is not using the EnumMember value to serialize the dictionary key.

Actual value (reproducible on Nuget version 2.5.0):
```
"account_filters": {
    "Depository": {
      "account_subtypes": [
        "checking"
      ]
    }
  }
```

The expected value should be:
```
"account_filters": {
    "depository": {
      "account_subtypes": [
        "checking"
      ]
    }
  }
```

**Comment by viceroypenguin** [comment]
Question about context. Are you serializing manually, or is this a bug in the connection to Plaid?

If you're serializing manually, then you will need to use `Going.Plaid.Converters.EnumConverter` to use the `EnumMember` attribute to serialize as expected. How to use this converter depends on the context of your application, so I would need more information to if you want assistance applying the converter. 

If this is a bug in the connection to Plaid, then let me know and I'll take a look next week when I'm off vacation. May need to do some work to convince the serializer to use the proper converter when applying a converter to the key type of a dictionary.

**Comment by grcodemonkey** [comment]
The bug appears to be in `Going.Plaid.Converters.EnumConverter`

If the Dictionary Key is an enum, it is simply doing the ToString() value of the enum vs using the `EnumMember` value.

I see code that is handling when TValue of a type `Dictionary<Tkey, TValue>` is an enum, however, the case where TKey is an enum is not handled.

**Comment by viceroypenguin** [comment]
Please check release 2.6.0-preview - it should resolve the issue. I'll do an official release of 2.6.0 next week. Thanks!

**Comment by grcodemonkey** [comment]
Works great thank you

**Comment by viceroypenguin** [comment]
Reopened until closed by merge

**Comment by viceroypenguin** [comment]
Released via v2.6.0
